test/playwright/utils/navigation.ts

```diff
@@ -89,8 +116,135 @@ export const currentContentType = async (page: Page) => {
+    )
+    await dismissTranslationBanner(page)
+  } else {
+    await page.goto(pathWithDir('/', dir))
+    await dismissTranslationBanner(page)
+    // Select the search type
+    if (searchType !== 'all') {
+      await page.click('[aria-label="All content"]')
+      await page.click(
+        `button[role="radio"]:has-text("${searchTypeNames[searchType]}")`
+      )
+    }
+    // Type search term
+    const searchInput = page.locator('main input[type="search"]')
+    await searchInput.type(term)
+    // Click search button
+    // Wait for navigation
+    await Promise.all([
+      page.click('[aria-label="Search"]'),
+      page.waitForNavigation(),
+    ])
+    await scrollDownAndUp(page)
+  }
+}
+
+/**
+ * Fills the search input in the page header, clicks on submit
+ * and waits for navigation.
+ */
+export const searchFromHeader = async (page: Page, term: string) => {
+  await page.fill('id=search-bar', term)
+  await Promise.all([
+    page.waitForNavigation(),
+    page.locator('button[type="submit"]').click(),
+  ])
+}
+
+/**
+ * Click on the first <mediaType> result: a link that contains
+ * /<mediaType>/ in its URL. We cannot use the 'startsWith' `^` matcher
+ * because localized routes start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstResult = async (page: Page, mediaType: MediaType) => {
+  await page.locator(`a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+/**
+ * Click on the first <mediaType> related result: a link that contains
+ * /<mediaType>/ in its URL in the 'aside' element for related media.
+ * We cannot use the 'startsWith' `^` matcher because localized routes
+ * start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstRelatedResult = async (
+  page: Page,
+  mediaType: MediaType
+) => {
+  await page.locator(`aside a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+export const scrollToBottom = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, document.body.scrollHeight)
+  })
+  await sleep(300)
+}
+
+export const scrollToTop = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, 0)
+  })
+  await sleep(200)
+}
+
+/**
+ * Used to load all lazy-loaded images in the page.
+ */
+export const scrollDownAndUp = async (page: Page) => {
+  await scrollToBottom(page)
+  await scrollToTop(page)
+}
+
+export const renderDirs = ['ltr', 'rtl'] as const
+
+export const renderModes = [
+  ['SSR', 'ltr'],
+  ['SSR', 'rtl'],
+  ['CSR', 'ltr'],
+  ['CSR', 'rtl'],
+] as const
+
+/**
+ * Adds '/ar' prefix to a rtl route. The path should start with '/'
+ */
+export const pathWithDir = (rawPath: string, dir: string) => {
+  const path = rawPath.startsWith('/') ? rawPath : `/${rawPath}`
+  return dir === 'rtl' ? `/ar${path}` : path
```

Comment:
We can have clearer types here (and remove the `startsWith` check using a template string type :slightly_smiling_face:)

```suggestion
export const pathWithDir = (path: `/${string}`, dir: 'ltr' | 'rtl') => {
  return dir === 'rtl' ? `/ar${path}` : path
```